packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -845,6 +845,18 @@ class RenderOpacity extends RenderProxyBox {
   @override
   bool get alwaysNeedsCompositing => child != null && (_alpha > 0);
 
+  @override
+  bool get isRepaintBoundary => alwaysNeedsCompositing;
```

Comment:
This is the tricky part. Everything is fine if we are always a repaint boundary, but that implies leaving this layer in the tree forever. There are reason why we don't want to do that, for example consider the degenerate case of Clip -> Opacity -> Transform -> Opacity -> Nothing.